ballerina/query_depth_validator.bal

```diff
@@ -67,8 +67,10 @@ class QueryDepthValidator{
 
     public isolated function visitSelection(parser:Selection selection, anydata data = ()) {
         if (selection.isFragment) {
-            parser:FragmentNode fragmentNode = self.documentNode.getFragments().get(selection.name);
-            self.visitFragment(fragmentNode);
+            if (self.documentNode.getFragments().keys().indexOf(selection.name) != ()) {
```

Comment:
Instead of this, can't we validate the fragments first? Let's move the `FragmentVisitor` initialization and validation to `Engine` and then validate fragments first.